org.code.toboggan.modelmgr/src/org/code/toboggan/modelmgr/extensions/project/ModelMgrProjectGrantPermissions.java

```diff
@@ -23,8 +25,12 @@ public void permissionGrantFailed(long projectID, String grantUsername, int perm
 
 	@Override
 	public void permissionsGrantedNotification(long projectID, String grantUsername, int permissionLevel) {
-		Permission permission = new Permission(grantUsername, permissionLevel, ss.getUsername(), new Date().toString());
-		pc.addPermission(projectID, grantUsername, permission);
+		if (ss.getProject(projectID) == null) {
```

Comment:
Permission granted notifications are fired in two distinct cases: the first is when we are just newly granted permission on a project (the if case above), and the second is when another person is granted permissions on a project that we already have (the else case below). The tests have example workflows for both of them.